nwb_conversion_tools/datainterfaces/ecephys/baserecordingextractorinterface.py

```diff
@@ -90,7 +92,13 @@ def subset_recording(self, stub_test: bool = False):
         if self.subset_channels is not None:
             kwargs.update(channel_ids=self.subset_channels)
 
-        recording_extractor = se.SubRecordingExtractor(self.recording_extractor, **kwargs)
+        if isinstance(self.recording_extractor, se.RecordingExtractor):
+            recording_extractor = se.SubRecordingExtractor(self.recording_extractor, **kwargs)
+        elif isinstance(self.recording_extractor, si.BaseRecording):
+            recording_extractor = self.recording_extractor.frame_slice(start_frame=0, end_frame=end_frame)
+        else:
+            raise f"{self.recording_extractor} should be either se.RecordingExtractor or si.BaseRecording"
```

Comment:
```suggestion
            raise ValueError(f"{self.recording_extractor} should be either se.RecordingExtractor or si.BaseRecording")
```
When raising an error, there are one of two approaches you can take:

(a) `assert "Some message."

which returns an `AssertionError` type, or

(b) `raise SomeError("Some message.")

which returns whatever error type you specify from the following list: https://docs.python.org/3/tutorial/errors.html

Reply:
Right, this is way more specific. Reading from the list I think this should be a type error though.

Reply:
I added this in a different commit.